pallene/coder.lua

```diff
@@ -309,7 +311,9 @@ end
 
 -- @returns the C parameter name for the upvalue u_id
 function Coder:c_upval(u_id)
-    return "uv" .. u_id
+    assert(self.current_func)
+    local typ = self.current_func.captured_vars[u_id].decl.typ
+    return lua_value(typ, string.format("&U[%d]", u_id))
```

Comment:
Did we accidentally delete the comment that justifies why it is safe to unbox the upvalue without a tag check?